### PR TITLE
Add except: ["inside-block"] to rule-empty-line-before

### PR DESCRIPTION
**Pull request by lutien**
> Which issue, if any, is this issue related to?

#2887

> Is there anything in the PR that needs further explanation?

Maybe this case needs more tests, feel free to point me.

### CODE REVIEW

File: lib/rules/rule-empty-line-before/README.md [hunk 1]
```diff
@@ -120,7 +120,7 @@ b {
 
 ## Optional secondary options
 
-### `except: ["after-rule", "after-single-line-comment", "inside-block-and-after-rule", "first-nested"]`
+### `except: ["after-rule", "after-single-line-comment", "inside-block-and-after-rule", "first-nested", "inside-block"]`
```
Comment (jeddy3) [inline lib/rules/rule-empty-line-before/README.md]: Options are listed alphabetically, can you please put this before `"first-nested"`...

Comment (jeddy3) [inline lib/rules/rule-empty-line-before/README.md]: Options are listed alphabetically, can you please put this before `"first-nested"`...

File: lib/rules/rule-empty-line-before/README.md [hunk 2]
```diff
@@ -217,6 +217,36 @@ The following patterns are *not* considered violations:
 }
 ```
 
+#### `"inside-block"`
```
Comment (jeddy3) [inline lib/rules/rule-empty-line-before/README.md]: Before `"first-nested"` please

**Comment by lutien** [comment]
@jeddy3  Hi! Do I need to do something more here?

File: lib/rules/rule-empty-line-before/index.js [hunk 3]
```diff
@@ -96,7 +97,8 @@ const rule = function(expectation, options, context) {
           isNested &&
           isAfterRule(rule)) ||
         (optionsMatches(options, "except", "after-single-line-comment") &&
-          isAfterSingleLineComment(rule))
+          isAfterSingleLineComment(rule)) ||
+        (optionsMatches(options, "except", "inside-block") && isNested)
```
Comment (CAYdenberg) [inline lib/rules/rule-empty-line-before/index.js]: Just move isNested to a new line to be constistent with the other conditions.

Comment (lutien) [reply lib/rules/rule-empty-line-before/index.js]: @CAYdenberg If I change it prettier fixes this back. I suppose this line is too short to break it.

Comment (CAYdenberg) [reply lib/rules/rule-empty-line-before/index.js]: Hm. Ok then. More of a personal preference that anything else.

**Comment by jeddy3** [comment]
> -   Added: `except: ["inside-block"]` option to `rule-empty-line-before` ([#2982](https://github.com/stylelint/stylelint/pull/2982)).

**Pull request merged**